### DISCUSSION TITLE
Qual caminho trilhar em 2018

### DISCUSSION THREAD

**Discussion by crisgon** [frontendbr/forum#912] [Discussão]
Pessoal, primeiramente, desculpe se esse tema já foi abordado aqui, mas é que dei uma pesquisada por vue nas issues e não vi algo do tipo.
Estou estudando vanilla JS e meu próximo passo vai ser ir pra algum framework. Vejo muita vaga de React e quase nenhuma de vue, mas vejo muito se falar do vue e me parece algo bem legal.
**Queria saber o que vocês acham que vale mais a pena dedicar os estudos em 2018 e por que?**
Acho que essa é uma das grandes dúvidas de muita gente.

**Comment by isaacguerreir** [comment]
React, porque é sem dúvida o que mais o mercado mais tem demanda no momento.

**Comment by lucasfcosta** [comment]
IMHO você deve estudar o que **você** gosta, não o que o mercado gosta. Até porque são as pessoas que moldam o mercado e não o contrário.

O mercado adota as tecnologias que os programadores consideram as melhores escolhas pra resolver um problema. **Not the other way round**.

Quanto mais curiosidade e dedicação você tiver em relação à um determinado tópico, melhor você vai ser. Foque em desenvolver suas skills de aprendizado, vá atrás do que você acha que é uma boa escolha e aprenda aquilo. Diversifique os tópicos, deixe que seu interesse te guie por onde você quiser seguir.

Frequentemente eu vejo pessoas perdendo a animação porque têm planos de estudos muito longos e fixos em vez de estudarem aquilo que elas mesmas acham interessante. Obviamente nem tudo vai ser um mar de rosas e alguns assuntos vão ser mais chatos que os outros, mas acho que se guiar pela própria paixão e curiosidade é essencial.

E eu não digo nem apenas se tratando de programação/tecnologia. Se você se interessar por psicanálise vá comprar um Lacan, linguística vá comprar Chomsky/Pinker, enfim, seja lá qual for o tópico: sociologia, antropologia, filosofia. Eu nunca me arrependi de ter aprendido nada na minha vida, eu simplesmente deixo a curiosidade me guiar.

Obviamente o mundo real tem constraints que não podem ser ignoradas: as pessoas precisam colocar comida na mesa. Mas acho que a balança sempre deve pesar mais pro que te interessa. Na long-run sempre tive muito mais retorno estudando o que eu curtia e indo atrás das coisas que eu achava legal do que tentando seguir cronogramas e planos "formais".

Eu odeio ser o cara que sempre usa o mesmo quote, mas eu realmente adoro esse aqui:

> Be curious. Read widely. Try new things. What people call intelligence just boils down to curiosity.
> Aaron Swartz

Mas enfim, isso aí tudo é só a famigerada Minha Opinião™.

Feel free to disagree.

Cheers,
Lucas

**Comment by fabriciofmsilva** [comment]
Acho que o @lucasfcosta mandou bem no comentário. Estudar o que gosta é fundamental para manter a motivação, inclusive quando entrar em algum tópico mais chato e tals, ou tentar se aprofundar em conceitos mais complexos.

Só tentando complementar um pouco, focando aí na sua pergunta, procure dar uma olhada nos principais frameworks e libs do mercado. Vai lá e estuda superficialmente cada um, aprenda os conceitos de cada, as motivações que levaram a ser criados. Depois se aprofunde no que mais gostar, ou naquele que vai te ajudar no projeto atual, ou na vaga que está aplicando.

Se estiver trabalhando num projeto, uma coisa bacana é se aprofundar nos conceitos e tecnologias do projeto, eu gosto muito de aprender sob demanda. Se no projeto o backend é em Java usando Spring, com banco de dados em MySQL e MongoDB, fazendo um cache em Redis, e ainda tem uma fila de mensageria em JMS, eu procuro estudar cada uma dessas tecnologias, principalmente os seus conceitos. Se eu não curto, tendo a ficar só no superficial, mas isso já me ajuda a ter uma visão mais macro/sistêmica. Que inclusive já me ajudou a tomar algumas decisões melhores no front-end.

Como você falou que tem estudado JS Vanilla, isso é ótimo, se gostar de JS continue se aprimorando com o Vanilla, frameworks e libs vem e vão. Estude também a base de progração, algoritmos, estrutura de dados, assim como também maneiras de resolver problemas, boas práticas, desing patterns, práticas de engenharia de software.

Seja como for, não fique só no seu domínio, parta para aprender além dele. Como falei anteriormente a resposta do Lucas foi bem completa. Se aprimore todo dia, aprende algo novo sempre!

**Comment by rubenmarcus** [comment]
sobre React, bem eu tenho sido entrevistado para vagas desde o começo de Janeiro e pelo que vi as vagas de Angular pagam MAIS que as de React... E empresas grandes tem pedido também...

Vue vi algumas.. Acho que esse mês passei por 20 entrevistas, e 10 eram de Angular.

**Comment by felipefialho** [comment]
Considero os SPAs muito parecidos entre si atualmente, significa que se você estudar e aprender um, a curva de aprendizado para aprender outras vai ser pequena.

Acredito que SPAs, PWAs e aplicações híbridas vão continuar em alta esse ano.

Mas sou total a favor da resposta do @lucasfcosta, é muito complicado manter concentração se não for um assunto do seu interesse.

**Comment by danielschmitz** [comment]
Acredito que compreendendo bem o JavaScript, incluindo es2016, Você pode ter uma base de conhecimento pequena em Angular, React e Vue. Tente saber, pelo menos, criar um todo list para cada uma delas, do zero, sabendo muito bem o que está fazendo. Então, na entrevista, o que a empresa pedir, você estará apto a fazer.  

tl;dr Saiba muito bem JS, o framework vc usa de acordo com a demanda

**Comment by leandroruel** [comment]
so vou deixar esse artigo aqui: https://medium.com/javascript-scene/top-javascript-libraries-tech-to-learn-in-2018-c38028e028e6
